### DISCUSSION TITLE
Testing & Fixtures: How to fill global fields with data?

### DISCUSSION THREAD

**Discussion by gddotorg** [craftcms/cms#11717] [Q&A] [answered]
Hi,

I am currently working on fixtures for our API tests. The Project Config already sets up a GlobalSet "Global options" with the necessary fields. Now I would like to fill out these fields with dummy data. As I understand the examples in the Craft core, GlobalSetFixture is intended for the setup, i.e. in my example the structure for "Global options". What I'm looking for, however, is a way to fill the global field "globalImprintPageText" with data. I tried using BaseElementFixture and EntryFixture, but my setup always failed. Does anyone have an idea?

**Comment by brandonkelly** [comment, marked as answer]
Currently there’s no way to set content to existing elements a la carte using fixtures.

The best way to do it would probably be from `_before()`, such as how we’re doing it from `UserElementTest`: (Although in your case you would be fetching the existing global set first, rather than creating a new one.)

https://github.com/craftcms/cms/blob/cf3f39924bddd1a69295b5af03dae2c1b03815ea/tests/unit/elements/UserElementTest.php#L314-L343
